net/http/http_network_session.cc

```diff
@@ -373,6 +373,25 @@ void HttpNetworkSession::DisableQuic() {
 void HttpNetworkSession::SetEnableQuic(bool enable_quic) {
   params_.enable_quic = enable_quic;
 }
+void HttpNetworkSession::SetEnableHttp2(bool enable_http2) {
+  if (params_.enable_http2 == enable_http2) {
+    return;
+  }
+  params_.enable_http2 = enable_http2;
+
+  if (params_.enable_http2) {
+    next_protos_.push_back(kProtoHTTP2);
+    if (base::FeatureList::IsEnabled(features::kAlpsForHttp2)) {
+      // Enable ALPS for HTTP/2 with empty data.
+      application_settings_[kProtoHTTP2] = {};
+    }
+  } else {
+    if (next_protos_.back() == kProtoHTTP2) {
```

Comment:
Is it guaranteed that HTTP2 is at the end of supported protocol list? Even for without our addition?

Reply:
Yes, it's only added here and in the constructor.